src/services/nnid/routes/people.ts

```diff
@@ -580,7 +580,8 @@ router.put('/@me', async (request: express.Request, response: express.Response):
 	const region: number = person.region ? person.region : pnid.region;
 	const countryCode: string = person.country ? person.country : pnid.country;
 	const language: string = person.language ? person.language : pnid.language;
-	const timezoneName: string = person.tz_name ? person.tz_name : pnid.timezone.name;
+	// The 3DS sends an empty object when the timezone isn't updated, so this fixes validation from failing.
+	const timezoneName: string = (person.tz_name && !!Object.keys(person.tz_name).length) ? person.tz_name : pnid.timezone.name;

```

Comment:
Can you clarify this a bit please?

> The 3DS sends an empty object

You mean instead of a *string* it sends an *empty object*?

Reply:
Correct, since it's being sent as xml, it just sends the tag like this: `<tz_name/>` instead of the full `<tz_name>timezone/name</tz_name>`. It doesn't seem to happen consistently, but not handling it would cause the app to hang, which is why I added this additional check.

Reply:
Gross. Alright then lol